doc_src/cmds/set.rst

```diff
@@ -35,6 +35,8 @@ The following options control variable scope:
 
 - ``-l`` or ``--local`` forces the specified shell variable to be given a scope that is local to the current block, even if a variable with the given name exists and is non-local
 
+- ``-f`` or ``--function`` makes the variable local to the current *function*. This is the same as defining it with ``--local`` outside of a block like ``begin`` or ``for``. When used outside of a function, it uses the top-level local scope instead.
```

Comment:
I suggest not defining function-scoped variables in terms of local variables; instead we should assign the complexity to `--local`, which we want to de-emphasize. That is, move the `--function` description above `--local` and don't use "local" in its description since that could be confusing.

My attempt:

> -f or --function scopes the variable to the currently executing function, or the top-level scope if no function is executing. The variable is erased when the function completes.
>
> -l or --local scopes the variable to the currently executing block. The variable is erased when the block completes. If there is no currently executing block, this is the same as `--function`.

Reply:
That's a brilliant idea, thanks!